### PR TITLE
adding query expansion

### PR DESCRIPTION
**Pull request by skamalas**
yep

### CODE REVIEW

**Comment by skamalas** [comment]
:+1:

**Comment by yahoocla** [comment]
CLA is valid!

File: evaluate.py [hunk 1]
```diff
@@ -144,7 +166,20 @@ def get_ap(Q, data, query_name, index_names, groundtruth_dir, ranked_dir=None):
...
+
+    # Whiten, and reduce dim of features
+    # Whitening is trained on the same images that we query against here for expediency
+    print 'Fitting PCA/whitening wth d=%d on %s ...' % (d, whiten_features)
+    _, whiten_params = run_feature_processing_pipeline(data, d=d)
+
+    return whiten_params
+
+
+def run_eval(queries_dir, groundtruth_dir, index_features, whiten_params, out_dir, agg_fn, qe_fn=None):
```
Comment (pumpikano) [inline evaluate.py:195]: @skamalas Can you update the docstring here?

File: evaluate.py [hunk 2]
```diff
@@ -215,7 +239,17 @@ def run_eval(queries_dir, groundtruth_dir, index_features, whiten_features, out_
     else:
         agg_fn = apply_ucrow_aggregation
 
-    mAP = run_eval(args.queries, args.groundtruth, args.index_features, args.whiten_features, args.out, agg_fn, args.d)
+    if args.qe > 0:
+        qe_fn = partial( simple_query_expansion, top_k=args.qe )
```
Comment (pumpikano) [inline evaluate.py]: No paren spaces

**Comment by skamalas** [comment]
I fixed it I think, wanna check it?

**Comment by pumpikano** [comment]
:+1: awesome

**Pull request merged**